Stratis.Bitcoin/BlockPulling/BlockPullerBehavior.cs

```diff
@@ -0,0 +1,175 @@
+        /// <param name="message">Received message.</param>
+        private void Node_MessageReceived(Node node, IncomingMessage message)
+        {
+            message.Message.IfPayloadIs<BlockPayload>((block) =>
+            {
+                block.Object.Header.CacheHashes();
+                this.QualityScore = Math.Min(BlockPuller.MaxQualityScore, this.QualityScore + 1);
+
+                uint256 blockHash = block.Object.Header.GetHash();
+                if (this.puller.DownloadTaskFinished(this, blockHash))
+                {
+                    foreach (Transaction tx in block.Object.Transactions)
+                        tx.CacheHashes();
+                    this.puller.PushBlock((int)message.Length, block.Object, this.cancellationToken.Token);
+                    this.AssignPendingVector();
+                }
+            });
+        }
+
+        /// <summary>
+        /// If there are any more blocks the node wants to download, this method assigns and starts 
+        /// a new download task for a specific peer node that this behavior represents.
+        /// </summary>
+        internal void AssignPendingVector()
+        {
+            Node attachedNode = this.AttachedNode;
+            if (attachedNode == null || attachedNode.State != NodeState.HandShaked || !this.puller.Requirements.Check(attachedNode.PeerVersion))
+                return;
+
+            uint256 block = null;
+            if (this.puller.AssignPendingDownloadTaskToPeer(this, out block))
+                attachedNode.SendMessageAsync(new GetDataPayload(new InventoryVector(attachedNode.AddSupportedOptions(InventoryType.MSG_BLOCK), block)));
+        }
+
+        /// <summary>
+        /// Sends a message to the connected peer requesting specific data.
+        /// </summary>
+        /// <param name="getDataPayload">Specification of the data to download - <see cref="GetDataPayload"/>.</param>
+        /// <remarks>Caller is responsible to add the puller to the map if necessary.</remarks>
+        internal void StartDownload(GetDataPayload getDataPayload)
+        {
+            Node attachedNode = this.AttachedNode;
+            if (attachedNode == null || attachedNode.State != NodeState.HandShaked || !this.puller.Requirements.Check(attachedNode.PeerVersion))
+                return;
+
+            foreach (InventoryVector inv in getDataPayload.Inventory)
+                inv.Type = attachedNode.AddSupportedOptions(inv.Type);
+
+            attachedNode.SendMessageAsync(getDataPayload);
+        }
+
+        /// <summary>
+        /// Connects the puller to the node and the chain so that the puller can start its work.
+        /// </summary>
+        protected override void AttachCore()
+        {
+            this.AttachedNode.MessageReceived += Node_MessageReceived;
+            this.ChainHeadersBehavior = this.AttachedNode.Behaviors.Find<ChainHeadersBehavior>();
+            AssignPendingVector();
```

Comment:
OK, that is new to me, I've been doing that XML doc PRs where I also fix code style everywhere, but this was not part of the code style (and as I say, MSVS does mark only those I mentioned). 

So this has to be a separate issue because it basically means we need to update code-style and then enforce it everywhere. Not that I would be happy about this particular one though, but OK. Still, I would consider that a separate issue and we should redesign the code-style policy heavily.